bin/generate_repos.py

```diff
@@ -115,23 +193,37 @@ def create_folder(folder):
         print ("exiting");
         sys.exit()
 
+
+# Load the git branch for autgrading from the course config file
+course_config_file = os.path.join('/var/local/submitty/courses/',
+                                  args.semester, args.course,
+                                  'config', 'config.json')
+with open(course_config_file) as open_file:
+    COURSE_JSON = json.load(open_file)
+course_git_autograding_branch = COURSE_JSON['course_details']['git_autograding_branch']
+# verify that the branch only contains alphabetic characters a-z
+if not re.match('^[a-z]+$',course_git_autograding_branch):
+    print (f"Invalid course git autograding branch '{course_git_autograding_branch}'")
+    course_git_autograding_branch = 'main'
```

Comment:
I really dislike the concept of this having an error, but continuing on anyway, especially given the amount of output generated here so an instructor might not even realize this happened. I very much would prefer a hard failure here for the instructor to fix the branch name to fit our requirements, e.g.:

```suggestion
    raise SystemError(f"Invalid course git autograding branch '{course_git_autograding_branch}'")
```

Reply:
generate repos is run automatically when a gradeable is created (for individual gradeables) and when each team forms (for team gradeables).  It is rarely needed to be directly run by instructors.   Propagating the failure to that UI would be alot more work.

Reply:
The only case instructors run it directly is if a students is added to the course after a gradeable was created.

Reply:
This field is currently only exposed to instructors via the file and has to be edited within the file? I suppose at some point we'll hook this up into the course settings page and so validation on that field could happen there.

Reply:
Yes, sure, could go on that page eventually.
I think this is a low priority feature.  I hope its mostly ignored.  
Everyone should switch to 'main'.  I just want to handle old courses, and have a way instructors with fragile infrastructure can continue using master (even in new courses) if necessary.

Reply:
I guess 🤷